src/Symfony/Component/Console/Application.php

```diff
@@ -1169,4 +1169,27 @@ private function splitStringByWidth($string, $width)
 
         return $lines;
     }
+
+    /**
+     * Returns all namespaces of the command name.
+     *
+     * @param string $name The full name of the command
+     *
+     * @return array The namespaces of the command
+     */
+    private function extractAllNamespaces($name)
+    {
+        $parts = explode(':', $name, -1);
```

Comment:
If we omit the `-1` in explode, it will return not only namespaces (`doctrine`, `doctrine:database`) but also the full command (`doctrine:database:create`). As this is also run for command aliases, it will return also full command aliases.

e.g. [this test](https://github.com/symfony/symfony/blob/c7bee8f861cc6f728e865f7e7b931c74a3302322/src/Symfony/Component/Console/Tests/ApplicationTest.php#L186) will fail, because instead of only `foo` as expected, `getNamespaces` will also return commands and aliases:

```
Array(
    0 => 'help'
    1 => 'list'
    2 => 'foo'
    3 => 'foo:bar'
    4 => 'afoobar'
    5 => 'foo:bar1'
    6 => 'afoobar1'
)
```

This third argument is equivalent to using `array_pop` after the explode, as the previously used function is doing: https://github.com/symfony/symfony/blob/c7bee8f861cc6f728e865f7e7b931c74a3302322/src/Symfony/Component/Console/Application.php#L1039